CONTRIBUTE.md

```diff
@@ -0,0 +1,333 @@
+} catch (err) {
+  console.log(err);
+  // expected output: RangeError: invalid language tag: EN_US
+}
+```
+
+## Format
+
+The aim is to try and fit examples into 12 lines of code, if it's not possible to create a useful example in 12 lines, it's fine to use more lines. We currently have examples of up to 25 lines.
+
+### Semi-colons
+
+There are valid arguments for and against using semi-colons. We use them.
+
+### Line Spacing
+
+To fit the space available we try and keep examples as compact as possible, using blank lines sparingly to increase readability.  
+
+One place we use a blank line is immediately preceding a `console.log` statement unless it is the first line in a block.
+
+For example:
+
+```
+construct: function(target, args) {
+  console.log("1 constructor called");
+  // expected output: " constructor called"
+
+  return new target(...args);
+}
+```
+and
+
+```
+var proxy1 = new Proxy(1, handler1);
+
+console.log(new proxy1("fierce").disposition);
+// expected output: "fierce"
+```
+
+### Indentation
+
+In order to keep things as concise as reasonably possible we indent with two spaces.
+
+### Instantiating an Object
+
+Most objects are instantiated in the normal way, for example:
+```
+var date1 = new Date();
+```
+
+The exception being `Object` in which case we instantiate like so:
+
+```
+var object1 = {};
+```
+
+### Instantiating an Array
+
+We instantiate arrays in the following way:
+
+```
+var array1 = [2, 5, 7, 9];
+```
+
+Note the spacing after the commas.
+
+### Passing parameters
+
+We space after commas when passing parameters to a function, but no space padding of parenthesis:
+
+```
+calcAngle(8, 10);
+```
+
+### Single Quotes
+
+We use Single quotes to denote strings:
+
+```
+setTimeout(resolve, 100, 'foo');
+```
+
+### Property Definition
+
+We chose clarity over brevity when defining object properties, for example:
+
+```
+var object1 = {
+  property1: 42,
+  property2: 'foo'
+};
+```
+
+Note the space after the colon.
+
+### Spaces between Operators
+
+For example:
+
+```
+var a = 1 + 2;
+
+if (a > b) {
+  return a;
+}
+```
+
+### Function Definition
+
+Generally functions are defined with the `function` keyword at the beginning:
+
+```
+function sum(a, b) {
+  return a + b;
+}
+```
+
+Note the space before the opening curly bracket.
+
+### Anonymous Function Definition
+
+Where ES6 predates the method or object we're illustrating we use the arrow function:
+
+```
+var sum = array1.reduce((a, b) => a + b);  
+```
+
+rather than:
+
+```
+var sum = array1.reduce(function(a, b) {
+  return a + b;  
+});
+```
+
+### Function Definition
+
+Named functions are defined like so:
+
+```
+function sum(a, b) {
+  return a + b;
+}
+```
+
+Note the function name starts with a lowercase letter and can be camelCased.
+
+### Class Definition
+
+As classes came in with ES6 we can use other ES6 concepts when defining them such as `const`. We also need to be aware of the changing scope of `this`.
```

Comment:
Nit: "As classes were introduced with ES6, we"

Reply:
👍